src/hooks/BookAppointment/useAppointmentSecondStep.hook.tsx

```diff
@@ -1,65 +1,116 @@
-import React, { useState, useEffect } from 'react';
-import { ListOfDoctors } from '@constants/mockData';
-
-interface Doctor {
-  id: string;
-  img: string;
-  name: string;
-  speciality: string;
-  located: string;
-  rating: string;
+import React, { useState, useMemo } from 'react';
+import { appointmentApi } from 'services/BookAppointmetService';
+
+interface Prop {
+  selectedDate: Date;
+  formattedTime: string;
+  setStep: React.Dispatch<React.SetStateAction<boolean>>;
+  specialization: number;
 }
 
-const useAppointmentSecondStepHook = () => {
-  const [filter, setFilter] = useState(``);
-  const [page, setPage] = useState(0);
-  const [allDoctors, setAllDoctors] = useState<Doctor[]>([]);
+const useAppointmentSecondStepHook = ({
+  selectedDate,
+  formattedTime,
+  setStep,
+  specialization,
+}: Prop) => {
+  const [limit, setLimit] = useState<number>(10);
+  const [selectedDoctor, setSelectedDoctor] = useState<number | null>(null);
+  const [filter, setFilter] = useState<string>(``);
+
+  // reverse time range for sending to backend
+  function reverseFormatTimeRange(timeRange: string) {
+    const [start, end] = timeRange
+      .split('-')
+      .map((time: string) => time.trim());
+
+    const startDate = new Date(selectedDate);
+    const endDate = new Date(selectedDate);
+
+    startDate.setHours(getHoursFromTime(start));
+    startDate.setMinutes(getMinutesFromTime(start));
+
+    endDate.setHours(getHoursFromTime(end));
+    endDate.setMinutes(getMinutesFromTime(end));
+
+    return {
+      start: startDate.toISOString(),
+      end: endDate.toISOString(),
+    };
+  }
+
+  function getHoursFromTime(time: string) {
+    const [hour] = time.split(':');
+    const isPM = time.includes('PM');
+    let formattedHour = parseInt(hour.trim());
+
+    if (formattedHour === 12) {
+      formattedHour = isPM ? formattedHour : 0;
+    } else {
+      formattedHour = isPM ? formattedHour + 12 : formattedHour;
+    }
```

Comment:
you have a lo of helpers to work with data, I think it make sense create folder helpers, and move such functions here in date.ts file